blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java

```diff
@@ -272,18 +272,7 @@ private Object convert(String name, ParameterValue v) throws IOException, Interr
         }
     }
     private boolean canSubmit(InputStep inputStep){
-
-        Authentication a = Jenkins.getAuthentication();
-        String submitter = inputStep.getSubmitter();
-        if (submitter==null || a.getName().equals(submitter)) {
-            return true;
-        }
-        for (GrantedAuthority ga : a.getAuthorities()) {
-            if (ga.getAuthority().equals(submitter)) {
-                return true;
-            }
-        }
-        return false;
+        return inputStep.canSubmit();
```

Comment:
Do you know for sure this honours the currently logged in users permissions somehow?